### PR TITLE
Preflight requests are not forwarded to services

### PR DESCRIPTION
**Pull request by sizief**
based on https://github.com/traefik/traefik/blob/master/pkg/middlewares/headers/header.go#L55 the preflight requests are not passed to services. I checked this also locally.

### More

- [ ] Added/updated tests
- [ ] Added/updated documentation

### CODE REVIEW

File: docs/content/middlewares/http/headers.md [hunk 1]
```diff
@@ -200,7 +200,7 @@ http:
 ### CORS Headers
 
 CORS (Cross-Origin Resource Sharing) headers can be added and configured in a manner similar to the custom headers above.
-This functionality allows for more advanced security features to quickly be set.
+This functionality allows for more advanced security features to quickly be set. If CORS headers are set, then the middleware does not pass preflight requests to any services, instead the response will be generated and will be send back to the user directly.
```
Comment (tomMoulard) [inline docs/content/middlewares/http/headers.md:203]:
```suggestion
This functionality allows for more advanced security features to quickly be set.
If CORS headers are set, then the middleware does not pass preflight requests to any services,
instead the response will be generated and sent back to the client directly.
```

File: docs/content/middlewares/http/headers.md [hunk 2]
```diff
@@ -201,6 +201,8 @@ http:
 
 CORS (Cross-Origin Resource Sharing) headers can be added and configured in a manner similar to the custom headers above.
 This functionality allows for more advanced security features to quickly be set.
+If CORS headers are set, then the middleware does not pass preflight requests to any services,
```
Comment (tomMoulard) [inline docs/content/middlewares/http/headers.md]:
```suggestion
If CORS headers are set, then the middleware does not pass preflight requests to any service,
```

**Review by kevinpollet — APPROVED** [review]
Thanks 👍

**Review by rtribotte — APPROVED** [review]
Thanks 👍

**Review by tomMoulard — APPROVED** [review]
LGTM :ok_hand:

**Pull request merged**